src/graphcast_agent/mod.rs

```diff
@@ -100,6 +100,13 @@ impl GraphcastAgentConfig {
         let boot_node_addresses = convert_to_multiaddrs(&boot_node_addresses.unwrap_or_default())
             .map_err(|_| GraphcastAgentError::ConvertMultiaddrError)?;
 
+        let mut discv5_enrs = discv5_enrs.unwrap_or_default();
+
+        if discv5_port.is_some() {
```

Comment:
No need to check for `discv5_port` here. `discv5` is always turned on and `discv5_port` is simply a matter of configuring to a port other than the waku default. [check](https://github.com/graphops/graphcast-sdk/blob/ad9c481a48b6eee4e128787cfd17f9a7e24b6506/src/graphcast_agent/waku_handling.rs#L187)